src/gc/gc.cpp

```diff
@@ -34934,7 +34962,7 @@ GCHeap::GarbageCollectGeneration (unsigned int gen, gc_reason reason)
         if (localCount != col_count)
         {
 #ifdef SYNCHRONIZATION_STATS
-            gc_lock_contended++;
+            Interlocked::Increment(&gc_lock_contended);
```

Comment:
I was using this as a shortcut for VolatileStore = VolatileLoad + 1.  even though there is no race, there is possibility of data loss as store buffer on one thread are not flushed.  But again sisnce thse are rare it is not really needed.